In [6]:
import pandas as pd

df = pd.read_csv("/titanic.csv")

df.head()

,survived,pclass,sex,age,sibsp,parch,fare,embarked,class,who,adult_male,deck,embark_town,alive,alone
0,0,3,male,22.0,1,0,7.2500,S,Third,man,True,NaN,Southampton,no,False
1,1,1,female,38.0,1,0,71.2833,C,First,woman,False,C,Cherbourg,yes,False
2,1,3,female,26.0,0,0,7.9250,S,Third,woman,False,NaN,Southampton,yes,True
3,1,1,female,35.0,1,0,53.1000,S,First,woman,False,C,Southampton,yes,False
4,0,3,male,35.0,0,0,8.0500,S,Third,man,True,NaN,Southampton,no,True


In [7]:
df[["age", "fare", "sibsp", "parch"]].describe()

,age,fare,sibsp,parch
count,714.000000,891.000000,891.000000,891.000000
mean,29.699118,32.204208,0.523008,0.381594
std,14.526497,49.693429,1.102743,0.806057
min,0.420000,0.000000,0.000000,0.000000
25%,20.125000,7.910400,0.000000,0.000000
50%,28.000000,14.454200,0.000000,0.000000
75%,38.000000,31.000000,1.000000,0.000000
max,80.000000,512.329200,8.000000,6.000000


In [8]:
df[["age", "fare", "sibsp", "parch"]].mode()

,age,fare,sibsp,parch
0,24.0,8.05,0,0


In [9]:
summary = pd.DataFrame({
    "Mean": df[["age", "fare", "sibsp", "parch"]].mean(),
    "Median": df[["age", "fare", "sibsp", "parch"]].median(),
    "Mode": df[["age", "fare", "sibsp", "parch"]].mode().iloc[0],
    "Std Deviation": df[["age", "fare", "sibsp", "parch"]].std(),
    "25th Percentile": df[["age", "fare", "sibsp", "parch"]].quantile(0.25),
    "50th Percentile": df[["age", "fare", "sibsp", "parch"]].quantile(0.50),
    "75th Percentile": df[["age", "fare", "sibsp", "parch"]].quantile(0.75)
})

summary

,Mean,Median,Mode,Std Deviation,25th Percentile,50th Percentile,75th Percentile
age,29.699118,28.0000,24.00,14.526497,20.1250,28.0000,38.0
fare,32.204208,14.4542,8.05,49.693429,7.9104,14.4542,31.0
sibsp,0.523008,0.0000,0.00,1.102743,0.0000,0.0000,1.0
parch,0.381594,0.0000,0.00,0.806057,0.0000,0.0000,0.0


In [10]:
summary.round(2)

,Mean,Median,Mode,Std Deviation,25th Percentile,50th Percentile,75th Percentile
age,29.70,28.00,24.00,14.53,20.12,28.00,38.0
fare,32.20,14.45,8.05,49.69,7.91,14.45,31.0
sibsp,0.52,0.00,0.00,1.10,0.00,0.00,1.0
parch,0.38,0.00,0.00,0.81,0.00,0.00,0.0


In [11]:
skewness = df[["age", "fare", "sibsp", "parch"]].skew()

skewness.round(2)

,0
age,0.39
fare,4.79
sibsp,3.70
parch,2.75


In [12]:
q1 = df[["age", "fare", "sibsp", "parch"]].quantile(0.25)
q3 = df[["age", "fare", "sibsp", "parch"]].quantile(0.75)

iqr = q3 - q1

iqr.round(2)

,0
age,17.88
fare,23.09
sibsp,1.00
parch,0.00


Descriptive Statistics Interpretation

Age: The mean age is about 29.70 years, while the median is 28 years. The mean is slightly higher than the median, indicating a small positive skew. The standard deviation of about 14.53 shows noticeable variation in passenger ages. The middle 50% of ages lie between approximately 20.13 and 38 years.

Fare: The mean fare is about 32.20, while the median is 14.45. The much higher mean and positive skewness indicate that the fare distribution is strongly right-skewed. The standard deviation of about 49.69 also shows high variability. The middle 50% of fares lie between approximately 7.91 and 31.

SibSp: The median and mode are both 0, meaning most passengers had no siblings or spouses aboard. The positive skewness indicates that a smaller number of passengers had higher SibSp values. The middle 50% of the data ranges from 0 to 1.

Parch: The median and mode are both 0, showing that most passengers had no parents or children aboard. The positive skewness indicates that a smaller number of passengers had higher Parch values. The IQR is 0 because both the 25th and 75th percentiles are 0.

Overall: Comparing the mean and median helps identify skewness, while standard deviation and the interquartile range help understand the spread of the data. Fare has particularly high variability and positive skewness, while Age has a more moderate distribution.